pandas/io/excel.py

```diff
@@ -788,6 +789,8 @@ def _conv_value(val):
         val = "{val}".format(val=val)
     elif is_list_like(val):
```

Comment:
this routine needs some re-factoring, something like

```def _conv_value(val):
   # doc-string here
   if is_integer(val):
        val = int(val)
    elif is_float(val):
        val = float(val)
    elif is_bool(val):
        val = bool(val)
   elif isinstance(val, (datetime, timedelta)):
        pass
    else:
       # comment here
        val = str(val)

    return val
```

with some additional tests

Reply:
This raises another issue. I take your meaning here since `datetime` and date are treated later on in write_cells, but `timedelta` is not explicitly dealt with. Turns out that writing `timedelta` is not supported when exporting .xls. Looks like this is an old issue (#9155).

I suggest not putting the `timedelta` case in `_conv_value`, since that case is handled in the openpyxl and xlsxwriter, and instead making the same conversion that xlsxwriter makes in the appropriate `write_cells`.

Reply:
@cbertinato yes this function actually could be moved to the top-level.